test/integration/verify.test.js

```diff
@@ -179,28 +199,75 @@ describe('verify batch content', () => {
     expect(files.filter(x => x === ORIGINAL_CTL_CHECKSUM_BLOB_NAME).length).toBe(1)
   })
 
-  test('throws error if batch file missing', async () => {
+  test('poll inbound does not throw error if batch file missing', async () => {
     await uploadBlob(ORIGINAL_CTL_BATCH_BLOB_NAME, EMPTY_CONTENT)
     await uploadBlob(ORIGINAL_CHECKSUM_BLOB_NAME, VALID_HASH)
     await uploadBlob(ORIGINAL_CTL_CHECKSUM_BLOB_NAME, EMPTY_CONTENT)
 
-    await expect(pollInbound()).rejects.toThrow()
+    await expect(pollInbound).not.toThrow()
   })
 
-  test('throws error if checksum file missing', async () => {
+  test('poll inbound does not throw error if checksum file missing', async () => {
     await uploadBlob(ORIGINAL_BATCH_BLOB_NAME, VALID_CONTENT)
     await uploadBlob(ORIGINAL_CTL_BATCH_BLOB_NAME, EMPTY_CONTENT)
     await uploadBlob(ORIGINAL_CTL_CHECKSUM_BLOB_NAME, EMPTY_CONTENT)
 
-    await expect(pollInbound()).rejects.toThrow()
+    await expect(pollInbound).not.toThrow()
   })
 
-  test('throws error if checksum control file missing', async () => {
+  test('poll inbound does not throw error if checksum control file missing', async () => {
     await uploadBlob(ORIGINAL_BATCH_BLOB_NAME, VALID_CONTENT)
     await uploadBlob(ORIGINAL_CTL_BATCH_BLOB_NAME, EMPTY_CONTENT)
     await uploadBlob(ORIGINAL_CHECKSUM_BLOB_NAME, VALID_HASH)
 
-    await expect(pollInbound()).rejects.toThrow()
+    await expect(pollInbound).not.toThrow()
+  })
+
+  test('verify batch throws error if batch file missing', async () => {
+    await uploadBlob(ORIGINAL_CTL_BATCH_BLOB_NAME, EMPTY_CONTENT)
+    await uploadBlob(ORIGINAL_CHECKSUM_BLOB_NAME, VALID_HASH)
+    await uploadBlob(ORIGINAL_CTL_CHECKSUM_BLOB_NAME, EMPTY_CONTENT)
+
+    await expect(verifyBatch).rejects.toThrow()
+  })
+
+  test('verify batch throws error if checksum file missing', async () => {
+    await uploadBlob(ORIGINAL_BATCH_BLOB_NAME, VALID_CONTENT)
+    await uploadBlob(ORIGINAL_CTL_BATCH_BLOB_NAME, EMPTY_CONTENT)
+    await uploadBlob(ORIGINAL_CTL_CHECKSUM_BLOB_NAME, EMPTY_CONTENT)
+
+    await expect(verifyBatch).rejects.toThrow()
+  })
+
+  test('verify batch throws error if checksum control file missing', async () => {
+    await uploadBlob(ORIGINAL_BATCH_BLOB_NAME, VALID_CONTENT)
+    await uploadBlob(ORIGINAL_CTL_BATCH_BLOB_NAME, EMPTY_CONTENT)
+    await uploadBlob(ORIGINAL_CHECKSUM_BLOB_NAME, VALID_HASH)
+
+    await expect(verifyBatch).rejects.toThrow()
+  })
```

Comment:
These three tests don't actually do anything.  They upload files, then check that calling verifyBatch results in an error, which it always will.  It doesn't actually call the `pollInbound` function.

But thinking about this, I think we've been going the wrong way about validating this behaviour.  This is an integration test, so we should be looking at outcomes, but we've been focusing on how can we check that `verifyBatch` throws an error.

If a file is missing, then what we should see if the original files still in the inbound folder.  So I think a simpler way of writing these tests would be, get rid of what I suggested earlier, ie the `verifyBatch` stuff above.

Then call poll inbound here and then get the files in the inbound folder the same way the tests below do and check that each of the three files uploaded are still there unchanged.

That seems a better test.

Reply:
I can take a look at doing that yeah - thanks